src/components/selected_card/SelectedCard.test.tsx

```diff
@@ -0,0 +1,190 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import { Provider } from 'react-redux';
+import { configureStore } from '@reduxjs/toolkit';
+import '@testing-library/jest-dom/vitest';
+import { vi } from 'vitest';
+import SelectedCard from './SelectedCard';
+import favoritesReducer, {
+  type FavoritePost,
+} from '../../store/features/favorites/favorites_slice';
+import * as downloadCsv from '../../helpers/downloadCsv';
+
+// Mock the downloadCsv module
+vi.mock('../../helpers/downloadCsv', () => ({
+  downloadFile: vi.fn(),
+}));
+
+const createTestStore = (
+  initialState: { favorites: FavoritePost[] } = { favorites: [] }
+) => {
+  return configureStore({
+    reducer: {
+      favorites: favoritesReducer,
+    },
+    preloadedState: {
+      favorites: initialState,
+    },
```

Comment:
instead of using original configureStore, could you mock it 

// Mock configureStore
vi.mock('@reduxjs/toolkit', async (importOriginal) => {
  const actual = await importOriginal() as any;
  return {
    ...actual,
    configureStore: vi.fn(),
  };
});